### ISSUE TITLE
White flash when loading prerendered page

### ISSUE THREAD

**Issue by nicolas-t** [chrisvfritz/prerender-spa-plugin#131]
Hello, 

When I load a prerendered page I first see the prerendered page, then the page becomes white again before being hydrated by vue.

At the end of the process my website is fully usable.

This is my HTML : 

```html
<!DOCTYPE html>
<html lang="en">
    <head>
    </head>
    <body>
        <div id="root"></div>
    </body>
</html>
```

And this how vue is being mounted: 

```javascript
const main = new Vue({
    store,
    router: Router,
    render () {
        return this.$createElement(Root)
    }
})

document.addEventListener("DOMContentLoaded", function(event) {
    main.$mount('#root')
})
```

The `Root` component also has a #root id.

This tweet makes me think I'm doing something wrong: 
https://twitter.com/youyuxi/status/910210352830349312

Do you think the white flash is the expecting behaviour ? Or is vue supposed to be smart and I'm doing something wrong ? 

Thanks

**Comment by nicolas-t** [comment]
It works with `data-server-rendered="true"` on my prerendered parent, like this: 
`<div id="root" data-server-rendered="true">`

Maybe this need to be added to the docs, or to the plugin itself.
After the docs `main.$mount('#root', true)` should do the job too, I haven't tried.

**Comment by stereobooster** [comment]
@nicolas-t can you provide an example of an application with white flash? I can not reproduce it with examples in this repository

**Comment by nicolas-t** [comment]
hello @stereobooster , can you give me your email so I can send you some examples ? thanks

**Comment by stereobooster** [comment]
my nickname @gmail.com

**Comment by nicolas-t** [comment]
@stereobooster  thanks, you can delete you message

**Comment by stereobooster** [comment]
actually this is not a secret 🤣 

```
git clone (any my repository)
git log
```

**Comment by stereobooster** [comment]
I found confirmation in [source code](https://github.com/vuejs/vue/blob/956756b1be7084daf8b6afb92ac0da7c24cde2a5/src/core/vdom/patch.js#L707)

```js
 if (oldVnode.nodeType === 1 && oldVnode.hasAttribute(SSR_ATTR)) {
  oldVnode.removeAttribute(SSR_ATTR)
  hydrating = true
}
```

```js
export const SSR_ATTR = 'data-server-rendered'
```

**Comment by nicolas-t** [comment]
Did you get my email ? :)

**Comment by stereobooster** [comment]
Yes I did. Later I discovered that this is not application specific, but framework specific. I thought reply here would be enough, sorry that I didn't respond to your email.

**Comment by nicolas-t** [comment]
No problem
